cmd/manager/main.go

```diff
@@ -70,6 +92,12 @@ func main() {
 		MetricsBindAddress:      *metricsBindAddress,
 		SyncPeriod:              syncPeriod,
 		Namespace:               farosflags.Namespace,
+		NewClient: func(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
```

Comment:
I've been looking through the available options here and it seems we can just provide `utils.NewRestMapper` as the `MapperProvider` and use the in-built utils to construct the clients, WDYT? Would result in a 1 line diff in this file 😄 

https://github.com/kubernetes-sigs/controller-runtime/blob/4276f3895df0acc9249f817eb86a47a3db6b7a9e/pkg/manager/manager.go#L96